### PR TITLE
DLS-8418: Adjustment Journey - Create add-small-producer Page

### PR DESCRIPTION
**Pull request by drewmiley**

### CODE REVIEW

File: app/controllers/correctReturn/AddASmallProducerController.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      val form: Form[AddASmallProducer] = formProvider(request.userAnswers)
+      val preparedForm = request.userAnswers.get(AddASmallProducerPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      Ok(view(preparedForm, mode))
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = controllerActions.withRequiredJourneyData(CorrectReturn).async {
```
Comment (ChrisHillCap) [inline app/controllers/correctReturn/AddASmallProducerController.scala:63]: Please check original variations flow, but for returns, we do many checks for the SDIL reference before saving it.

File: app/models/correctReturn/AddASmallProducer.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import models.SmallProducer
+import play.api.libs.json._
+
+case class AddASmallProducer(producerName:Option[String], referenceNumber:String, lowBand: Long, highBand: Long)
+
+object AddASmallProducer {
+  implicit val format = Json.format[AddASmallProducer]
+
+  def toSmallProducer(addASmallProducer: AddASmallProducer): SmallProducer =
```
Comment (ChrisHillCap) [inline app/models/correctReturn/AddASmallProducer.scala:27]: also we don't do this in returns, is the validation different?

File: app/views/helpers/lowAndHighBandInputs.scala.html [hunk 3]
```diff
@@ -29,10 +29,14 @@
         suffix = Some(PrefixOrSuffix(
             content = Text("litres")
         )),
-        errorMessage = if (form("lowBand").hasErrors) {
-            Some(ErrorMessage(content = Text(messages(s"${form.errors.head.message}"))))
-        } else None,
-        value = form.data.get("lowBand")
+        errorMessage =
```
Comment (ChrisHillCap) [inline app/views/helpers/lowAndHighBandInputs.scala.html:32]: why was this chnged for lowband, and also hint is not a message key

**Review by ChrisHillCap — CHANGES_REQUESTED** [review]
some accessibility issues on returns need amending for this page as its going to be the same problems for this page

[DLS-8479](https://jira.tools.tax.service.gov.uk/browse/DLS-8479)
unsure whether this has been done [DLS-8477](https://jira.tools.tax.service.gov.uk/browse/DLS-8477)
[DLS-8481](https://jira.tools.tax.service.gov.uk/browse/DLS-8481)
[DLS-8454](https://jira.tools.tax.service.gov.uk/browse/DLS-8454)

File: app/views/helpers/lowAndHighBandInputs.scala.html (hunk 3 above)
Comment (drewmiley) [reply app/views/helpers/lowAndHighBandInputs.scala.html:32]: This is the fix for https://jira.tools.tax.service.gov.uk/browse/DLS-8480
I'm not sure what you mean by 'also hint is not a message key'

File: app/models/correctReturn/AddASmallProducer.scala (hunk 2 above)
Comment (drewmiley) [reply app/models/correctReturn/AddASmallProducer.scala:27]: No - it just seems a neater way of converting between models than doing it in the controller

**Comment by drewmiley** [comment]
[DLS-8477](https://jira.tools.tax.service.gov.uk/browse/DLS-8477) has been fixed in this ticket

File: app/views/helpers/lowAndHighBandInputs.scala.html (hunk 3 above)
Comment (ChrisHillCap) [reply app/views/helpers/lowAndHighBandInputs.scala.html:32]: visuallyHiddenText = Some(messages("Error"))
should be 
visuallyHiddenText = Some(messages("messagekeyfrommessagefile"))

Comment (ChrisHillCap) [reply app/views/helpers/lowAndHighBandInputs.scala.html:32]: does this need to be changed for highband aswell

**Comment by drewmiley** [comment]
Screenshot 2023-07-26 at 13 36 18

**Comment by drewmiley** [comment]
Screenshot 2023-07-26 at 13 41 12

**Comment by drewmiley** [comment]
Screenshot 2023-07-26 at 13 41 31

File: app/views/helpers/lowAndHighBandInputs.scala.html (hunk 3 above)
Comment (drewmiley) [reply app/views/helpers/lowAndHighBandInputs.scala.html:32]: Fixed

File: app/controllers/correctReturn/AddASmallProducerController.scala (hunk 1 above)
Comment (drewmiley) [reply app/controllers/correctReturn/AddASmallProducerController.scala:63]: Fixes for duplicate SDIL ref. Cannot currently do check for `sdilConnector.checkSmallProducerStatus` as there is no `request.returnPeriod`

Comment (ChrisHillCap) [reply app/controllers/correctReturn/AddASmallProducerController.scala:63]: there is a return period from the select controller which should be used

Comment (drewmiley) [reply app/controllers/correctReturn/AddASmallProducerController.scala:65]: Fixed

File: conf/messages.en [hunk 4]
```diff
@@ -27,6 +27,7 @@ timeout.keepAlive = Stay signed in
 timeout.signOut = Sign out
 
 litres.lowBand = Litres in the low band
+litres.lowBand.visuallyHidden = Error
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply conf/messages.en:30]: This was originally how I wrote it, but it was changed to fit in changes I had to rebase on top of. Very happy to change it if you would like me to?

[the remark this answers is not in the document]
Comment (adamliptrot-oc) [reply conf/messages.en:30]: The main thing is the colon as it adds in a pause when read out. Other stuff is just for ease of maintenance.

[the remark this answers is not in the document]
Comment (drewmiley) [reply conf/messages.en:30]: I believe this is in https://github.com/hmrc/play-frontend-hmrc/blob/main/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukErrorMessage.scala.html not our code?

**Comment by drewmiley** [comment]
Need to add ViewSpec

File: conf/messages.en (hunk 4 above)
[the remark this answers is not in the document]
Comment (drewmiley) [reply conf/messages.en:30]: I'll change the other back to `site.error = Error` though

**Review by ginnyamazed — APPROVED** [review]

**Pull request merged**